src/v2/Apps/Partner/Routes/Artists/PartnerArtists.tsx

```diff
@@ -0,0 +1,136 @@
+import { Box, Text, Button, Flex } from "@artsy/palette"
+import React, { useEffect, useRef, useState } from "react"
+import {
+  createPaginationContainer,
+  graphql,
+  RelayPaginationProp,
+} from "react-relay"
+import { PartnerArtists_partner } from "v2/__generated__/PartnerArtists_partner.graphql"
+import {
+  PartnerArtistListFragmentContainer as PartnerArtistList,
+  PartnerArtistListPlaceholder,
+} from "../../Components/PartnerArtists"
+
+const PAGE_SIZE = 20
+
+export interface ArtistsRouteProps {
+  partner: PartnerArtists_partner
+  relay: RelayPaginationProp
+}
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+export const ArtistsRoute: React.FC<ArtistsRouteProps> = ({
+  partner: { artists, distinguishRepresentedArtists, slug },
+  relay,
+}) => {
+  const [artistsLoaded, setArtistsLoaded] = useState(!relay.hasMore())
+  const errCounter = useRef(0)
+
+  const refetchArtists = () => {
+    setArtistsLoaded(false)
+    errCounter.current = 0
+
+    loadMoreArtists()
+  }
+
+  const loadMoreArtists = () => {
+    relay.loadMore(PAGE_SIZE, async error => {
+      if (error) {
+        console.error(error)
+        errCounter.current += 1
+        // Wait before next try.
+        await sleep(500)
+      } else {
+        errCounter.current = 0
+      }
+
+      if (errCounter.current >= 3) {
+        setArtistsLoaded(true)
+
+        return
+      }
+
+      if (relay.hasMore()) {
+        loadMoreArtists()
+      } else {
+        setArtistsLoaded(true)
+      }
+    })
+  }
+
+  useEffect(() => {
+    if (relay.hasMore()) {
+      loadMoreArtists()
+    }
+  }, [])
+
+  return (
+    <Box mt={4}>
+      <Text variant="title" mb={6}>
+        Artists
+      </Text>
+      {artistsLoaded ? (
+        <>
+          <PartnerArtistList
+            partnerSlug={slug}
+            artists={artists.edges}
+            distinguishRepresentedArtists={distinguishRepresentedArtists}
+          />
+          {errCounter.current > 0 && (
+            <Flex flexDirection="column" mt={2} alignItems="center">
+              <Button variant="secondaryOutline" onClick={refetchArtists}>
+                Load more artists
+              </Button>
+            </Flex>
+          )}
+        </>
+      ) : (
+        <PartnerArtistListPlaceholder />
```

Comment:
Rendering the entire placeholder on a page fetch seems really disturbing. We'd probably want to render the loading state of the button and simply append rather than flashing the whole thing with a loading state.

Reply:
Do you want to get rid of skeleton placeholder at all or just use the loading state of the button on refetch and show already loaded artists?

Reply:
It seems like in this case just the loading state of the button should be fine. We have no way of knowing how many are going to get appended and we only really use the skeleton style of loader to prevent layout shifts on content that's loaded strictly client-side.

Reply:
Done.

Loading without error: 
![ezgif com-gif-maker (8)](https://user-images.githubusercontent.com/79979820/113696307-80085a00-96da-11eb-9fb9-a0d52b3731f1.gif)

Loading with error: 
![ezgif com-gif-maker (9)](https://user-images.githubusercontent.com/79979820/113696730-0755cd80-96db-11eb-964c-5a6d06158c56.gif)